sdk/nodejs/automation/cmd.ts

```diff
@@ -37,26 +46,140 @@ export class CommandResult {
     }
 }
 
-const unknownErrCode = -2;
+export interface PulumiCommandOptions {
+    version?: semver.SemVer;
+    root?: string;
+    /**
+     * Skips the minimum CLI version check, see `PULUMI_AUTOMATION_API_SKIP_VERSION_CHECK`.
+     * @internal
+     */
+    skipVersionCheck?: boolean;
+}
 
-/** @internal */
-export async function runPulumiCmd(
+export class PulumiCommand {
+    private constructor(readonly command: string, readonly version: semver.SemVer | null, readonly root?: string) {}
+
+    /**
+     * Get a new Pulumi instance that uses the installation in `opts.root`.
+     * Defaults to using the pulumi binary found in $PATH if no installation
+     * root is specified.  If `opts.version` is specified, it validates that
+     * the CLI is compatible with the requested version and throws an error if
+     * not.
+     */
+    static async get(opts?: PulumiCommandOptions): Promise<PulumiCommand> {
+        const command = opts?.root ? path.resolve(path.join(opts.root, "bin/pulumi")) : "pulumi";
```

Comment:
If you path.resolve both sides of this ternary then we can just assume `command` is always a full path and not need to pass `root` to `PulumiCommand`.